DNN Platform/DotNetNuke.Abstractions/Settings/ISettingsService.cs

```diff
@@ -0,0 +1,148 @@
+        T Get<T>(string key)
+            where T : IConvertible;
+
+        /// <summary>
+        /// Gets the encrypted value converted to the generic type.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The type to convert the retrieved value to.
+        /// This should implement <see cref="IConvertible"/>.
+        /// </typeparam>
+        /// <param name="key">The key to retrieve.</param>
+        /// <returns>The decrypted value of the retrieved key.</returns>
+        T GetEncrypted<T>(string key)
+            where T : IConvertible;
+
+        /// <summary>
+        /// Gets the encrypted value converted to the generic type.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The type to convert the retrieved value to.
+        /// This should implement <see cref="IConvertible"/>.
+        /// </typeparam>
+        /// <param name="key">The key to retrieve.</param>
+        /// <param name="passPhrase">The passpPhrase to use for decrypting the value.</param>
+        /// <returns>The decrypted value of the retrieved key.</returns>
+        T GetEncrypted<T>(string key, string passPhrase)
+            where T : IConvertible;
+
+        /// <summary>
+        /// Gets the encrypted value as a string.
+        /// </summary>
+        /// <param name="key">The key to retrieve.</param>
+        /// <returns>The decrypted value of the retrieved key.</returns>
+        string GetEncrypted(string key);
+
+        /// <summary>
+        /// Gets the encrypted value as a string.
+        /// </summary>
+        /// <param name="key">The key to retrieve.</param>
+        /// <param name="passPhrase">The passpPhrase to use for decrypting the value.</param>
+        /// <returns>The decrypted value of the retrieved key.</returns>
+        string GetEncrypted(string key, string passPhrase);
+
+        /// <summary>
+        /// Deletes the setting.
+        /// </summary>
+        /// <param name="key">The setting key to delete.</param>
+        void Delete(string key);
+
+        /// <summary>
+        /// Deletes the setting.
+        /// </summary>
+        /// <param name="key">The setting key to delete.</param>
+        /// <param name="clearCache">Clears the cache if true.</param>
+        void Delete(string key, bool clearCache);
+
+        /// <summary>
+        /// Deletes all settings and clears the cache.
+        /// </summary>
+        void DeleteAll();
+
+        /// <summary>
+        /// Deletes all settings.
+        /// </summary>
+        /// <param name="clearCache">If true the cache will be cleared.</param>
+        void DeleteAll(bool clearCache);
+
+        /// <summary>
+        /// Adds or updates the setting for a specified key.
+        /// </summary>
+        /// <param name="key">The setting key to update.</param>
+        /// <param name="value">The value to set.</param>
+        void Update(string key, string value);
```

Comment:
Do we want generic Update APIs to match the generic Get APIs?

Reply:
Should the update API support full C# objects that get serialized into json or should it just support primitives such as
- string
- int
- double
- float
- long
- short
- bool

Reply:
[The `SerializationController` exposes an overload that takes a custom serializer](https://github.com/dnnsoftware/Dnn.Platform/blob/80f8a4dcb57ca63d2b4a7fedc01c7be000e500f9/DNN%20Platform/Library/Entities/Modules/Settings/SerializationController.cs#L20). otherwise it [checks for `DateTime` and `TimeSpan` and finally falls back to `Convert.ToString`](https://github.com/dnnsoftware/Dnn.Platform/blob/80f8a4dcb57ca63d2b4a7fedc01c7be000e500f9/DNN%20Platform/Library/Entities/Modules/Settings/SerializationController.cs#L131-L146).

Reply:
WOW! That sounds really powerful, I will certainly swap out this implementation for that. Should we migrate that to the abstractions project as part of this PR. That sounds like something really useful to be available via Dependency Injection

Reply:
If you're up for it, that would make sense to me.  We can also do that as a separate PR if it makes sense to reduce the scope of this one.

Reply:
I'll submit a new PR for the `SerializationController`

Reply:
Summited PR #4087